### PR TITLE
feat: add functions-js

### PR DESCRIPTION
**Pull request by thorwebdev**
Bring the Func(🕺)

### CODE REVIEW

**Review by inian — APPROVED** [review]

**Comment by cursecodes** [comment]
It's happening...

File: src/SupabaseClient.ts [hunk 1]
```diff
@@ -71,6 +73,15 @@ export default class SupabaseClient {
...
     this.authUrl = `${_supabaseUrl}/auth/v1`
     this.storageUrl = `${_supabaseUrl}/storage/v1`
+
+    const isPlatform = _supabaseUrl.match(/(supabase\.co)|(supabase\.in)/)
+    if (isPlatform) {
+      const urlParts = _supabaseUrl.split('.')
+      this.functionsUrl = `${urlParts[0]}.functions.${urlParts[1]}.${urlParts[2]}`
+    } else {
+      this.functionsUrl = `${_supabaseUrl}/functions/v1`
+    }
```
Comment (dshukertjr) [inline src/SupabaseClient.ts:83]: @thorwebdev What is the purpose of this check right here? I was implemenntinng functions in Dart SDK, and was wondering.

Comment (soedirgo) [reply src/SupabaseClient.ts:83]: This is to make local development using the CLI work.

Hosted:
- `SUPABASE_URL` is `https://{ref}.supabase.co`
- Function URL is `https://{ref}.functions.supabase.co/{function-name}`

Local:
- `SUPABASE_URL` is `http://localhost:{api-port}`
- Function URL is `http://localhost:{api-port}/functions/v1`

Comment (dshukertjr) [reply src/SupabaseClient.ts:83]: Thanks @soedirgo !

**Pull request merged**